test/e2e-olm/elasticsearch_test.go

```diff
@@ -30,175 +31,172 @@ func TestElasticsearchCluster(t *testing.T) {
 }
 
 func singleNodeTest(t *testing.T) {
-	f := test.Global
-
-	ctx := test.NewContext(t)
-	namespace, err := ctx.GetWatchNamespace()
-	if err != nil {
-		t.Fatal(err)
-	}
+	namespace := operatorNamespace
 	t.Logf("Found namespace: %v", namespace)
 
 	esUUID := utils.GenerateUUID()
 	t.Logf("Using UUID for elasticsearch CR: %v", esUUID)
 
-	if err = createElasticsearchSecret(t, f, ctx, esUUID); err != nil {
+	if err := createElasticsearchSecret(t, k8sClient, esUUID); err != nil {
 		t.Fatal(err)
 	}
 
 	dataUUID := utils.GenerateUUID()
 	t.Logf("Using GenUUID for data nodes: %v", dataUUID)
 
 	// Create CR with a single node with client, data and master roles
-	cr, err := createElasticsearchCR(t, f, ctx, esUUID, dataUUID, 1)
+	_, err := createElasticsearchCR(t, k8sClient, esUUID, dataUUID, 1)
 	if err != nil {
 		t.Fatalf("could not create exampleElasticsearch: %v", err)
 	}
 
 	dplName := fmt.Sprintf("elasticsearch-%v-cdm-%v-1", esUUID, dataUUID)
-	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, dplName, 1, retryInterval, timeout)
+	err = utils.WaitForDeployment(t, k8sClient, namespace, dplName, 1, retryInterval, timeout)
 	if err != nil {
 		t.Fatalf("timed out waiting for first node deployment %v: %v", dplName, err)
 	}
 
-	ctx.Cleanup()
-	_ = e2eutil.WaitForDeletion(t, f.Client.Client, cr, cleanupRetryInterval, cleanupTimeout)
-	t.Log("Finished successfully")
+	cleanupEsTest(t, k8sClient, namespace, esUUID, dataUUID)
+	t.Log("SingleNodeTest finished successfully")
 }
 
-func multipleNodesTest(t *testing.T) {
-	f := test.Global
+func cleanupEsTest(t *testing.T, cli client.Client, namespace string, esUUID string, dataUUID string) {
+	esSecret := &corev1.Secret{}
+	key := types.NamespacedName{Name: elasticsearchNameFor(esUUID), Namespace: namespace}
+	err := waitForDeleteObject(t, k8sClient, key, esSecret, retryInterval, timeout)
+	if err != nil {
+		t.Errorf("cannot remove es secret: %v", err)
+	}
 
-	ctx := test.NewContext(t)
-	namespace, err := ctx.GetWatchNamespace()
+	es := &loggingv1.Elasticsearch{}
+	err = waitForDeleteObject(t, k8sClient, key, es, retryInterval, timeout)
 	if err != nil {
-		t.Fatal(err)
+		t.Errorf("cannot remove elasticsearch CR: %v", err)
 	}
-	t.Logf("Found namespace: %v", namespace)
 
+	// wait for ES deployment to be removed
+	for i := 0; i < 2; i++ {
+		name := fmt.Sprintf("elasticsearch-%v-cdm-%v-%d", esUUID, dataUUID, i+1)
+		key = types.NamespacedName{Name: name, Namespace: operatorNamespace}
+		esDeployment := &apps.Deployment{}
+		err = wait.Poll(cleanupRetryInterval, cleanupTimeout, func() (done bool, err error) {
+			err = k8sClient.Get(context.Background(), key, esDeployment)
+			if err != nil && errors.IsNotFound(err) {
+				return true, nil
+			}
+			fmt.Printf("Wait for cleaning up elasticsearch deployment %s\n", key.String())
+			return false, nil
+		})
+		if err != nil {
+			t.Error(err)
+		}
+	}
```

Comment:
Thinking of this a little bit more, we don't need to wait for deployments to be deleted. The work is done by the k8s GC, since we delete the `elasticsearch` CR.

Reply:
Right. I changed to checking the status of the elasticsearch CR because somehow in the my local cluster the e2e failed between tests due to unchanged status after deleting the CR.